### PR TITLE
MINOR: Kafka Streams docs fixes

### PR DESCRIPTION
**Pull request by mjsax**
- add missing section to TOC
- add default value for client.id

### CODE REVIEW

**Review by lucasbru — APPROVED** [review]

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -97,6 +97,7 @@
             <ul>
               <li><a class="reference internal" href="#naming" id="id17">Naming</a></li>
               <li><a class="reference internal" href="#default-values" id="id18">Default Values</a></li>
+              <li><a class="reference internal" href="#parameters-controlled-by-kafka-streams" id="id26">Default Values</a></li>
```
Comment (cadonna) [inline docs/streams/developer-guide/config-streams.html]: I suppose this should be
```suggestion
              <li><a class="reference internal" href="#parameters-controlled-by-kafka-streams" id="id26">Parameters controlled by Kafka Streams</a></li>
```

**Review by cadonna — COMMENTED** [review]
Thanks @mjsax !

I have one question

File: docs/streams/developer-guide/config-streams.html (hunk 1 above)
Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html]: Thanks. Great catch. C&p error.

**Review by cadonna — APPROVED** [review]
LGTM!

**Pull request merged**